pkg/oam/util/helper.go

```diff
@@ -0,0 +1,109 @@
+package util
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	plur "github.com/gertd/go-pluralize"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core/v1alpha2"
+)
+
+const (
+	KindDeployment = "Deployment"
+	KindService    = "Service"
+)
+
+// FetchWorkloadDefinition fetch corresponding workloadDefinition given a workload
+func FetchWorkloadDefinition(ctx context.Context, r client.Reader, workload *unstructured.Unstructured) (
+	[]*unstructured.Unstructured, error) {
+	// The name of the workloadDefinition CR is the CRD name of the component which is <purals>.<group>
+	gvr := getGVResource(workload.Object)
+	wldName := gvr.Resource + "." + gvr.Group
+	nn := types.NamespacedName{Name: wldName}
+	// Fetch the corresponding workloadDefinition CR
+	workloadDefinition := &v1alpha2.WorkloadDefinition{}
+	if err := r.Get(ctx, nn, workloadDefinition); err != nil {
+		return nil, err
+	}
+	return fetchChildResources(ctx, r, workload, workloadDefinition.Spec.ChildResourceKinds)
+}
+
+func fetchChildResources(ctx context.Context, r client.Reader, workload *unstructured.Unstructured,
```

Comment:
Looks like this depends on https://github.com/crossplane/oam-kubernetes-runtime/pull/20

Reply:
I've removed that part of code here, partly because the test is not "conform" to that repo.